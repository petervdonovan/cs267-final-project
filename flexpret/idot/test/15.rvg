[mu []
[def (A) [matrix 4 by 4 at t0]]
[[file {Reset_Handler} {
  Reset_Handler:
    [id init]
    [single t5
    {
      [A matrix-allocate]
      [iterate {row} of [[A matrix-rows] a0 a1 a2]
      {
        [iterate {addr} of [map [row t1 t2] mapaddr]
        {
          li t3 42
          sw t3 [id addr]
        }]
      }]
      [iterate {row} of [[A matrix-rows] a0 a1 a2]
      {
        [iterate {addr} of [map [row t1 t2] mapaddr]
        {
          lw a3 [id addr]
          [sim-print a3 t4]
        }]
      }]
    }]
    rdcycle a4
    [single-core a2 [critical-hart t5 a1 {
      [get-hartid t6]
      [sim-print t6 a0]
      [sim-print a4 a0]
    }]]
    [id die]}]]
]
