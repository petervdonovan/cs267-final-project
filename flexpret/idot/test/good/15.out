Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:         42
Core-0:          0
Core-0:        277
Core-0:          1
Core-0:        277
Core-0:          2
Core-0:        277
Core-0:          3
Core-0:        277
Core-0:          4
Core-0:        277
Core-0:          5
Core-0:        277
Core-0:          6
Core-0:        277
Core-0:          7
Core-0:        277
Core-0 is done
Core-1 is done
Core-2 is done
Core-3 is done
Core-4 is done
Core-5 is done
Core-6 is done
Core-7 is done
Core-8 is done
Core-9 is done
Core-10 is done
Core-11 is done
Core-12 is done
Core-13 is done
Core-14 is done
Core-15 is done
All cores are done terminating
Assertion failed: Program terminated sucessfully
